src/openstudio_lib/OSDocument.cpp

```diff
@@ -1444,14 +1468,18 @@ bool OSDocument::saveAs() {
 
     // saves the model to modelTempDir / filePath.filename()
     // also copies the temp files to user location
+    LOG(Debug, "Saving " << modelPath << " starting");
     bool saved = saveModel(this->model(), modelPath, toPath(m_modelTempDir));
-    OS_ASSERT(saved);
-
-    this->setSavePath(toQString(modelPath));
-
-    this->markAsUnmodified();
-
-    fileSaved = true;
+    if (saved) {
+      LOG(Debug, "Saving " << modelPath << " complete");
+      this->setSavePath(toQString(modelPath));
+      this->markAsUnmodified();
+      fileSaved = true;
+    } else {
+      LOG(Debug, "Saving " << modelPath << " failed");
+      QMessageBox::warning(this->mainWindow(), tr("Failed to save model"),
+                           tr("Failed to save model, make sure that you do not have the location open and that you have correct write access."));
+    }
```

Comment:
Noted, that's the actual change that might have an impact. The rest is just `LOG(Debug, ...)`